src/libcore/macros.rs

```diff
@@ -337,21 +337,16 @@ macro_rules! try {
 
 /// Write formatted data into a buffer
 ///
-/// This macro accepts a 'writer' (any value with a `write_fmt` method), a format string, and a
-/// list of arguments to format.
+/// This macro accepts a format string, a list of arguments, and a 'writer'.
+/// Arguments will be formatted according to the specified format string and the result will be
+/// passed to the writer.
+/// The writer may be any value with a `write_fmt` method; generally this comes from an
+/// implementation of either the [`std::fmt::Write`][fmt_write] or the [`std::io::Write`][io_write]
```

Comment:
I know that you're adapting existing text here, but while you're here... the current style is to do

    [`std::fmt::Write`]

in the text and then

    [`std::fmt::Write`]: URL

for the link. Could you change all of these to be in that style? Thank you!